src/shared/observables/createOptionStream.ts

```diff
@@ -7,14 +7,15 @@ import { Options } from '../options';
 import { vscode } from '../../vscodeAdapter';
 import { Observable, Observer } from 'rxjs';
 import { publishBehavior } from 'rxjs/operators';
+import { csharpDevkitExtensionId } from '../../utils/getCSharpDevKit';
 
 export default function createOptionStream(vscode: vscode): Observable<Options> {
     return Observable.create((observer: Observer<Options>) => {
         const disposable = vscode.workspace.onDidChangeConfiguration((e) => {
             //if the omnisharp or csharp configuration are affected only then read the options
             if (
                 e.affectsConfiguration('dotnet') ||
-                e.affectsConfiguration('omnisharp') ||
+                (!vscode.extensions.getExtension(csharpDevkitExtensionId) && e.affectsConfiguration('omnisharp')) ||
```

Comment:
💡 Unless we know `getExtension` is very efficient, consider reversing this condition